### PR TITLE
Fix crashes in ssh-tunnel code.

### PR DESCRIPTION
**Pull request by cjcullen**
Also add a metric to count how often we fail to open tunnels.

### CODE REVIEW

**Comment by cjcullen** [comment]
@brendandburns @a-robinson

File: pkg/master/master.go [hunk 1]
```diff
@@ -844,31 +846,22 @@ func (m *Master) refreshTunnels(user, keyfile string) error {
...
 	// Refresh loop for tunnels
 	// TODO: could make this more controller-ish
-	go func() {
-		for {
-			time.Sleep(5 * time.Minute)
-			if err := m.refreshTunnels(user, keyfile); err != nil {
-				glog.Errorf("Failed to refresh SSH Tunnels: %v", err)
-			}
+	go util.Until(func() {
+		time.Sleep(5 * time.Minute)
```
Comment (a-robinson) [inline pkg/master/master.go]: Why not move this sleep into the util.Until call?

File: pkg/master/master.go [hunk 2]
```diff
@@ -844,31 +846,22 @@ func (m *Master) refreshTunnels(user, keyfile string) error {
...
-	go func() {
-		for {
-			if err := m.loadTunnels(user, keyfile); err != nil {
-				glog.Errorf("Failed to load SSH Tunnels: %v", err)
-			}
-			var sleep time.Duration
-			if len(m.tunnels) == 0 {
-				sleep = time.Second
-			} else {
-				// tunnels could lag behind current set of nodes
```
Comment (a-robinson) [inline pkg/master/master.go]: Can you keep this comment around? Even with it I can't really pick out what the meaning of each case is

File: pkg/master/master.go (hunk 1 above)
Comment (cjcullen) [inline pkg/master/master.go]: util.Until doesn't allow me to sleep before my func(), which is what I want to do here.

**Comment by a-robinson** [comment]
LGTM

File: pkg/master/master.go (hunk 2 above)
Comment (cjcullen) [inline pkg/master/master.go]: Sorry. It's back.

**Comment by brendandburns** [comment]
Merging this, as I want to include it in 0.19.2

**Pull request merged**